lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua

```diff
@@ -228,8 +228,13 @@ end
 --------------------------------------------------------
 -- Other
 --------------------------------------------------------
-function EGP:ValidEGP( Ent )
-	return IsValid( Ent ) and (Ent:GetClass() == "gmod_wire_egp" or Ent:GetClass() == "gmod_wire_egp_hud" or Ent:GetClass() == "gmod_wire_egp_emitter")
+do
+	local GetTable = FindMetaTable( "Entity" ).GetTable
```

Comment:
Can you add `local IsValid = FindMetaTable( "Entity" ).IsValid`? I'm also curious if this function returns false if nil is passed into it or if it errors.

Reply:
If it errors, then replace `IsValid( ent )` with `ent and IsValid( ent )`

Reply:
It handles `nil`:
```
lua_run print( FindMetaTable( "Entity" ).IsValid( nil ) )
> print( FindMetaTable( "Entity" ).IsValid( nil ) )...
false
```

Applied 👍